buildlib/cbuild

```diff
@@ -474,17 +478,17 @@ def run_rpm_build(args,spec_file,env):
             print >> F,"""
 import os,subprocess;
 with open("/etc/passwd","a") as F:
-   print >> F, {passwd!r};
+   F.write({passwd!r} + "\\n");
 with open("/etc/group","a") as F:
-   print >> F, {group!r};
+   F.write({group!r} + "\\n");
 os.setgid({gid:d});
 os.setuid({uid:d});
 
 # Get RPM to tell us the expected tar filename.
 for ln in subprocess.check_output(["rpmspec","-P",{tspec_file!r}]).splitlines():
-   if ln.startswith("Source:"):
-      tarfn = ln.strip().partition(' ')[2].strip();
-os.symlink({tarfn!r},os.path.join("SOURCES",tarfn));
+   if ln.startswith(b"Source:"):
+      tarfn = ln.strip().partition(b' ')[2].strip();
+os.symlink({tarfn!r},os.path.join(b"SOURCES",tarfn));
```

Comment:
Where is this b coming from and why ?

Reply:
subprocess.check_output returns a bytes, so 'ln' is a bytes, 'tarfn' is a bytes, etc.

Reply:
oh ok. Sorry I'm a total python noobie.
This cleared it for me:
https://stackoverflow.com/questions/6269765/what-does-the-b-character-do-in-front-of-a-string-literal.

I'm good with the first patch then :)

Reply:
Oh, I thought you were asking something else.. Yes, the need for the 'b'  is related to very invasive unicode handling in python3. The two alternates here are to use the 'b'  or to call 'decode()' early on. The latter suffers from a couple of subtle faults if the spec file has non-ASCII characters, while 'b' works properly in all cases, if not a little ugly.

Reply:
I hate python....

Reply:
It's better to use `decode()` to get proper strings.

Reply:
@bdrung, yes that is what I thought was being asked.. There are subtle faults with using decode - rpmbuild doesn't do this so we shouldn't either.

Reply:
@jgunthorpe Isn't the file to be in a certain format? e.g. most files in `debian/` needs to be UTF-8. Then `decode()` is the right choice.